tests/test_end_to_end.py

```diff
@@ -67,3 +76,40 @@ def test_xarray_zarr_subpath(
 
     ds = xr.open_dataset(os.path.join(tmp_target_url, "subpath"), engine="zarr")
     xr.testing.assert_equal(ds.load(), daily_xarray_dataset)
+
+
+# @pytest.mark.parametrize("target_chunks", [{"time": 1}, {"time": 2}, {"time": 3}])
+def test_reference(
+    daily_xarray_dataset, netcdf_local_file_pattern_sequential, pipeline, tmp_target_url
+):
+    pattern = netcdf_local_file_pattern_sequential
+
+    with pipeline as p:
+        (
+            p
+            | beam.Create(pattern.items())
+            | OpenURLWithFSSpec()
+            | OpenWithKerchunk(file_type=pattern.file_type)
+            | DropKeys()
+            | CombineReferences(concat_dims=["time"], identical_dims=["lat", "lon"])
+            # TODO: variablize file_type to test parquet as well.
+            # FIXME: `WriteCombinedReference` probably needs to share an interface with
+            # `StoreToZarr`, in order for `pangeo-forge-runner` to know how to dynamically
+            # inject target_root argument.
+            | WriteCombinedReference(target=tmp_target_url, reference_file_type="json")
+        )
+    # NOTE: tmp_target_url is a directory ending in .zarr; maybe change that.
+    full_path = tmp_target_url + "/target.json"
+    of = fsspec.get_mapper("reference://", fo=full_path)
+    ds = xr.open_dataset(of, engine="zarr", backend_kwargs={"consolidated": False})
+
+    # FIXME: this assertion fails due to NaT in first postion of ds.time, and also
+    # int -> float variable dtype casting in ds. Maybe some other issues too.
```

Comment:
That's usually a sign that the data underlying a time coordinate are not correct and are not properly decoding to a datetime.

My recommendation for debugging is to bypass xarray and open the target store directly with `zarr.open_group`. Then look and see if you are able to actually load the data from the arrays with the kerchunk reference mapper. It may be the case that the reference json file itself is readable, but the way it is referring the the netCDF files is not correct.

Reply:
Thanks @rabernat. @norlandrhagen and I can look into this together tomorrow.

Reply:
Turns out @norlandrhagen's hunch from yesterday was correct! This was fixed by simply upgrading to the latest release of kerchunk>=0.1.0. No other debugging needed. 😄